index.bs

```diff
@@ -16,13 +16,439 @@ WPT Display: closed
+    </ol>
+</div>
+
+Integration with Fetch Metadata {#fetch-metadata-integration}
+==============================================
+
+The [:Sec-Fetch-Storage-Access:] header is appended to outgoing requests alongside other Fetch Metadata headers. [[!FETCH-METADATA]] Modify the definition of [$append the Fetch metadata headers for a request$] by inserting the following as step 6:
+
+<div algorithm="append the Sec-Fetch-Storage-Access header">
+    6.  <a lt='set-storage-access'>Set the `Sec-Fetch-Storage-Access` header</a> for <var ignore>r</var>.
+</div>
+
+Integration with Fetch {#fetch-integration}
+===========================================
+
+Handling these headers requires modifications to a few different parts of Fetch. [[!FETCH]]
+
+## `Origin` header ## {#origin-header-integration}
+
+When making a decision on whether to retry a request and force it to include cross-site cookies, a server ought to be informed as to the initiator of the request. I.e., the request ought to include the [:Origin:] header whenever it also includes the `Sec-Fetch-Storage-Access: inactive` header. Modify the definition of [=append a request Origin header=] by rewriting step 4 as:
+
+<div algorithm="modified append a request Origin header">
+    4. If at least one of the following conditions is true:
+         * |request|'s [=request/storage access status=] is "<code>[=storage access status/inactive=]</code>" and |request|'s [=request/credentials mode=] is "`include`"
+         * |request|'s [=request/method=] is neither \``GET`\` nor \``HEAD`\`
+
+        Then:
+</div>
+
+The rest of the algorithm is unmodified.
+
+## HTTP-fetch ## {#http-fetch}
+
+Insert a new step after step 5 in [=HTTP fetch=]:
+
+<div algorithm="modified HTTP-fetch">
+    6. If the result of [=perform a storage access retry check|performing a storage access retry check=] for <var ignore>request</var> is success, then return the result of running [=HTTP-storage-access-retry-fetch=] given <var ignore>fetchParams</var>.
+</div>
+
+Insert a new step after step 6.1 (before "switch on request's redirect mode"):
+
+<div algorithm="modified HTTP-fetch 2">
+    2. Set <var ignore>request</var>'s [=request/single-hop cache mode=] to null.
+</div>
+
+The rest of the algorithm is unmodified.
+
+<div algorithm>
+    To <dfn export>HTTP-storage-access-retry-fetch</dfn> given a [=fetch params=] |fetchParams|, run the following steps:
+
+    1. Let |request| be |fetchParams|'s [=request=].
+    1. Assert: |request|'s [=request/storage access status=] [=string/is=] "<code>[=storage access status/inactive=]</code>".
+    1. Assert: |request|'s [=request/eligible for storage-access=] is false.
+    1. If |request|'s [=request/redirect count=] is 20, then return a [=network error=].
+    1. Increase |request|'s [=request/redirect count=] by 1.
+    1. [=list/Append=] |request|'s [=request/url=] to |request|'s [=request/URL list=].
+    1. Set |request|'s [=request/single-hop cache mode=] to "`reload`".
```

Comment:
I'm not 100% sure I understand what the cache mode change does, but I trust you discussed this with Dom or someone else who knows what they're doing 😅

Reply:
Aah, yeah, I think that makes sense.